recipes-kernel/kernel-tests/kernel-tests-files/kernel_dmesg_diff.py

```diff
@@ -232,49 +242,38 @@ def parse_args():
+                             'Should be of the format "2023-03-30 15:32:43.203476". Date strings for previous dmesg logs can be found in the output of previous runs of this '\
+                             'test or can be extracted from the Mongo database using a viewer tool like Compass. This flag is useful for debugging issues with this test.')
+    parser.add_argument('--skip_upload', help='Skip upload of dmesg record to database. Useful when debugging.', action="store_true")
+    parser.add_argument('--suppress_diff_output', help='Reduces noise by removing the dmesg log diff from the test output. Useful when debugging.', action="store_true")
+    return parser.parse_args()
 
 logger = Logger()
 args = parse_args()
 db = DB(args.server, args.user, args.password)
 
-if args.get_current_log_from_db:
-    previous_query = {}
-    kernel_version = KernelVersion(args.get_current_log_kernel_version)
-    previous_query['kernel_version'] = { '$lt': kernel_version.version_dict }
-    previous_query['device_desc'] = args.get_current_log_device_desc
-    previous_query['kernel_version.major'] = kernel_version.major
-    previous_query['kernel_version.minor'] = kernel_version.minor
-    previous_query['os_version_major_minor'] = args.get_current_log_os_version
-
-    current_query = copy.copy(previous_query)
-    dmesg_log = get_old_dmesg_log(db, logger, current_query, kernel_version, "current")
-    assert dmesg_log, "Could not find matching current log from database."
-
-    # Add +1 to the prerelease version so the query (which looks for an earlier version) will return the version we actually want.
-    kernel_version.version_dict["prerelease"] += 1
-    old_dmesg_log = get_old_dmesg_log(db, logger, previous_query, kernel_version)
+if args.current_log_db_date:
+    current_dmesg_record = get_dmesg_record_by_date(db, args.current_log_db_date, logger)
+    assert current_dmesg_record, "Could not find matching current log record from database."
+    current_dmesg_log = strip_headers(current_dmesg_record['dmesg_log'])
+
+    previous_dmesg_record = get_previous_dmesg_record(db, KernelVersion(current_dmesg_record['kernel_version_full']), current_dmesg_record['os_version_major_minor'], current_dmesg_record['device_desc'], logger)
 else:
-    dmesg_log = get_dmesg_log()
-    old_dmesg_log = get_old_dmesg_log(db, logger)
+    current_dmesg_log = get_dmesg_log()
+
+    kernel_version = KernelVersion()
+    os_version = OsVersion()
+    device_desc = get_device_desc()
+    previous_dmesg_record = get_previous_dmesg_record(db, kernel_version, os_version.major_minor, device_desc, logger)
+
+previous_dmesg_log = strip_headers(previous_dmesg_record['dmesg_log']) if previous_dmesg_record else ''
 
-if not args.get_current_log_from_db:
-    upload_log(db, dmesg_log, logger)
+if not args.current_log_db_date and not args.skip_upload:
```

Comment:
nitpick: it appears that logs aren't uploaded not only when `skip_upload` is set but also when `current_log_db_date` is set. Maybe mention this in `current_log_db_date`'s help?

Reply:
Added a note to this effect and made a couple other small tweaks to the command line help.